### PR TITLE
DLP: Added sample for inspect image for sensitive data

### PR DESCRIPTION
**Pull request by jainamtrivedi-crest**
- Added sample for [Inspect Image for Sensitive Data](https://cloud.google.com/dlp/docs/samples/dlp-inspect-image-listed-infotypes) and test cases

### CODE REVIEW

**Review by annadata — APPROVED** [review]
LGTM.

File: dlp/api/Snippets/InspectImageForSensitiveDataWithListedInfoTypes.cs [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+                Data = ByteString.FromStream(new FileStream(filePath, FileMode.Open))
+            }
+        };
+
+        // Specify the type of info the inspection will look for.
+        var infoTypes = new InfoType[]
+        {
+            new InfoType { Name = "PHONE_NUMBER" },
+            new InfoType { Name = "EMAIL_ADDRESS" },
+            new InfoType { Name = "CREDIT_CARD_NUMBER" }
```
[the remark this answers is not in the document]
Comment (jainamtrivedi-crest) [reply dlp/api/Snippets/InspectImageForSensitiveDataWithListedInfoTypes.cs]: Done. Thanks.

**Comment by jainamtrivedi-crest** [comment]
CI is getting failed because the issue where buckets were not getting deleted properly in create and update stored infoType test cases. Raised a fix PR: https://github.com/GoogleCloudPlatform/dotnet-docs-samples/pull/2320

File: dlp/api/Snippets.Tests/UpdateStoredInfoTypesTests.cs [hunk 2]
```diff
@@ -64,8 +64,13 @@ public void TestUpdate()
...
                 // Delete the created buckets.
-                storage.DeleteBucket(bucketOutput.Name, new DeleteBucketOptions { DeleteObjects = true });
-                storage.DeleteBucket(bucket.Name, new DeleteBucketOptions { DeleteObjects = true });
+                storage.DeleteBucket(bucketOutput.Name);
+                var objects = storage.ListObjects(bucket.Name);
+                foreach (var ob in objects)
+                {
+                    await storage.DeleteObjectAsync(bucket.Name, ob.Name);
+                }
+                await storage.DeleteBucketAsync(bucket.Name);
```
Comment (amanda-tarafa) [inline dlp/api/Snippets.Tests/UpdateStoredInfoTypesTests.cs:73]: Please go back to using `DeleteObjects = true` as that should definitely work here. And if it doesn't let me know as soon as it happens (do not rerun tests so I can see the failure).

**Review by amanda-tarafa — COMMENTED** [review]
Looks good except for the changes in delete bucket. See my note there.

File: dlp/api/Snippets.Tests/UpdateStoredInfoTypesTests.cs (hunk 2 above)
Comment (amanda-tarafa) [reply dlp/api/Snippets.Tests/UpdateStoredInfoTypesTests.cs:73]: Alternatively consider using RandomBucketFixture and BucketCollector here the same as it's done in #2305

Comment (jainamtrivedi-crest) [reply dlp/api/Snippets.Tests/UpdateStoredInfoTypesTests.cs]: RandomBucketFixture seems to be working fine. I did not get any errors while running it on local. So, I have changed the sample with RandomBucketFixture. Thanks.

File: dlp/api/Snippets.Tests/UpdateStoredInfoTypesTests.cs [hunk 3]
```diff
@@ -28,45 +28,40 @@ public class UpdateStoredInfoTypesTests : IClassFixture<DlpTestFixture>
...
+            using var randomOutputBucketFixture = new RandomBucketFixture();
+            var outputBucketName = randomOutputBucketFixture.BucketName;
+            using var outputBucketCollector = new BucketCollector(outputBucketName);
 
-            try
-            {
-                // Create Stored InfoType first.
-                var storedInfoTypeId = $"github-usernames-update-{random.Next()}";
-                var response = CreateStoredInfoTypes.Create(_fixture.ProjectId, $"gs://{bucket.Name}", storedInfoTypeId);
+            Random random = new Random();
```
Comment (amanda-tarafa) [inline dlp/api/Snippets.Tests/UpdateStoredInfoTypesTests.cs:46]:
```suggestion
```

**Review by amanda-tarafa — COMMENTED** [review]
Thanks, looks good. Just remove a tiny change.

File: dlp/api/Snippets.Tests/UpdateStoredInfoTypesTests.cs (hunk 3 above)
Comment (jainamtrivedi-crest) [reply dlp/api/Snippets.Tests/UpdateStoredInfoTypesTests.cs:46]: We are appending some random number after the name of Stored InfoType. We are going to need this.

Comment (amanda-tarafa) [reply dlp/api/Snippets.Tests/UpdateStoredInfoTypesTests.cs:46]: Oh sorry, didn't see that. Thought it was just leftover from bucket creation.

**Review by amanda-tarafa — APPROVED** [review]

**Pull request merged**